lib/aruco_detector.hpp

```diff
@@ -6,52 +6,73 @@
 #include "classification_model.hpp"
 
 struct ArucoParams {
-    std::vector<int> ids;
-    std::vector<std::vector<cv::Point2f>> corners;
-    std::vector<cv::Vec3d> rvecs;
-    std::vector<cv::Vec3d> tvecs;
+    cv::Mat camera_mat;
+    cv::Mat distcoeffs;
+    cv::Ptr<cv::aruco::Dictionary> dictionary;
+    float markersize;
+
+    ArucoParams()
+            : camera_mat((cv::Mat_<double>(3, 3) << 216.357407, 0, 214.624283,
+            0, 216.357407, 115.447563,
+            0, 0, 1)),
+              distcoeffs(cv::Mat::zeros(5, 1, CV_64F)),
+              dictionary(cv::aruco::getPredefinedDictionary(cv::aruco::DICT_4X4_50)),
+              markersize(0.15) {}
 };
 
-class Aruco_Detector {
+class ArucoDetector {
 public:
-    Aruco_Detector(const cv::Mat& cameraMatrix, int dictionaryType, float markerSize = 0.15)
-        : cameraMatrix(cameraMatrix), dictionaryType(dictionaryType), markerSize(markerSize) {}
-
-    friend ClassificationModel::Detection model_classify(Aruco_Detector &detector, cv::Mat& image, float threshold) {
-        ArucoParams params;
-        if (detector.detectMarkers(image, params)) {
-            return ClassificationModel::Detection::ARUCO;
-        } else {
-            return ClassificationModel::Detection::NONE;
-        }
-    }
+    ArucoDetector(const ArucoParams& arucoParams)
+            : m_aruco_params(arucoParams) {}
 
-    cv::Rect get_bounding_box() {
-        return cv::Rect();
+    // Changed member variables to match ArucoParams
+    friend ClassificationModel::Detection model_classify(ArucoDetector &detector, cv::Mat& image, float threshold);
+    friend cv::Rect ModelGetBoundingBox(const ArucoDetector& detector);
+
+    cv::Rect model_get_bounding_box() {
+        assert(!m_detection_result.ids.empty() && "No markers detected");
+        std::vector<cv::Point2f>& corners = m_detection_result.corners[0];
+        cv::Rect boundingBox = cv::boundingRect(corners);
+        return boundingBox;
     }
 
-private:
-    cv::Mat cameraMatrix;
-    int dictionaryType;
-    float markerSize;
-    ArUcoDetectionResult detectionResult;
+    std::pair<cv::Vec3d, cv::Vec3d> get_pose() {
+        assert(!m_detection_result.ids.empty() && "No markers detected");
+        cv::Vec3d rvec = m_detection_result.rvecs[0];
+        cv::Vec3d tvec = m_detection_result.tvecs[0];
+        return std::make_pair(rvec, tvec);
+    }
 
-    bool detectMarkers(cv::Mat& image, ArucoParams& params) {
-        cv::Mat distCoeffs = cv::Mat::zeros(5, 1, CV_64F);
-        cv::Ptr<cv::aruco::Dictionary> dictionary = cv::aruco::getPredefinedDictionary(dictionaryType);
+    bool DetectMarkers(cv::Mat& image) {
+        std::vector<int> ids;
+        std::vector<std::vector<cv::Point2f>> corners;
+        std::vector<cv::Vec3d> rvecs, tvecs;
 
         if (image.empty()) {
             std::cerr << "Error: Input image is empty." << std::endl;
             return false;
         }
```

Comment:
There is a separate logging library (`spdlog`) that is used to report errors like these: https://github.com/kknives/michi/blob/fd030796cc2bcc88a81b5c568af13e30245abc91/lib/yolov8_arrow.hpp#L159